core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -374,7 +375,9 @@ memories_instantiate(const WASMModule *module, WASMModuleInstance *module_inst,
     for (i = 0; i < module->import_memory_count; i++, import++, memory++) {
         uint32 num_bytes_per_page = import->u.memory.num_bytes_per_page;
         uint32 init_page_count = import->u.memory.init_page_count;
-        uint32 max_page_count = import->u.memory.max_page_count;
+        uint32 max_page_count = max_memory_pages == 0
+                                    ? import->u.memory.max_page_count
+                                    : max_memory_pages;
```

Comment:
max_page_count should be no smaller than init_page_count, had better add a check like comment in aot_runtime.c?